src/main/java/ru/hogwarts/school/service/FacultyService.java

```diff
@@ -0,0 +1,45 @@
+package ru.hogwarts.school.service;
+
+import org.springframework.stereotype.Service;
+import ru.hogwarts.school.model.Faculty;
+import ru.hogwarts.school.model.Student;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class FacultyService {
+
+    private final HashMap<Long, Faculty> faculties = new HashMap<>();
+    private long lastId = 0;
+
+    public Faculty createFaculty(Faculty faculty){
+        faculty.setId(++lastId);
+        faculties.put(lastId,faculty);
+        return faculty;
+    }
+    public Faculty findFaculty(long id){
+        return faculties.get(id);
+    }
+    public Faculty editFaculty(Faculty faculty){
+        if(faculties.containsKey(faculty.getId())){
+            faculties.put(faculty.getId(),faculty);
+            return faculty;
+        }
+        return null;
+    }
+    public Faculty deleteFaculty(long id){
+        return faculties.remove(id);
+    }
+    public Collection<Faculty> getAllFaculties(){
+        return faculties.values();
+    }
+    public List<Faculty> sortByColour(String colour){
```

Comment:
sort обычно мы используем для указания, что метод делает сортировку. Если мы фильтруем данные, то лучше использовать слово filter
```suggestion
    public List<Faculty> filterByColour(String colour){
```